server/task_executor.go

```diff
@@ -158,8 +161,31 @@ func runMigration(ctx context.Context, l *zap.Logger, server *Server, task *api.
 		return true, nil, err
 	}
 
+	// For tenant mode project, we will only write back latest schema file on the last task.
+	tenantWriteBack := true
+	if issue != nil {
+		project, err := server.composeProjectByID(ctx, task.Database.ProjectID)
+		if err != nil {
+			return true, nil, err
+		}
+		if project.TenantMode == api.TenantModeTenant {
+			var lastTask *api.Task
+			for i := len(issue.Pipeline.StageList) - 1; i >= 0; i-- {
+				stage := issue.Pipeline.StageList[i]
+				if len(stage.TaskList) > 0 {
+					lastTask = stage.TaskList[len(stage.TaskList)-1]
+					break
+				}
+			}
+			// Not the last task yet.
+			if lastTask != nil && task.ID != lastTask.ID {
+				tenantWriteBack = false
+			}
+		}
+	}
+
 	// If VCS based and schema path template is specified, then we will write back the latest schema file after migration.
-	if vcsPushEvent != nil && repository.SchemaPathTemplate != "" {
+	if vcsPushEvent != nil && repository.SchemaPathTemplate != "" && tenantWriteBack {
```

Comment:
Here is another optimization.

We only need considering writeback if vcsPushEvent != nil && repository.SchemaPathTemplate != "".

So these 2 can be in the outside block.

and we then we have an internal parameter like writeback to determine if we do need writeback.

Reply:
Done